### PR TITLE
Add hover provider sample plugin

### PR DESCRIPTION
**Pull request by akurinnoy**
Signed-off-by: Oleksii Kurinnyi <okurinny@redhat.com>

### Demo:

hover-provider-plugin

### CODE REVIEW

File: hover-provider-plugin/LICENSE [hunk 1]
```diff
@@ -0,0 +1,81 @@
+Eclipse Public License - v 2.0
```
Comment (benoitf) [inline hover-provider-plugin/LICENSE]: formatting of the license is stange
compare with :
https://www.eclipse.org/org/documents/epl-2.0/EPL-2.0.txt

File: hover-provider-plugin/src/hover-provider-frontend-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*********************************************************************
+* Copyright (c) 2018 undefined
```
Comment (benoitf) [inline hover-provider-plugin/src/hover-provider-frontend-plugin.ts]: undefined ?

File: hover-provider-plugin/yarn.lock [hunk 3]
```diff
@@ -0,0 +1,2777 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (benoitf) [inline hover-provider-plugin/yarn.lock]: using @next with this yarn.lock implies that we're fixing the package in all case

File: hover-provider-plugin/LICENSE (hunk 1 above)
Comment (akurinnoy) [reply hover-provider-plugin/LICENSE]: Thanks, I'll fix it here and in PR for `che-generator-plugin`.

File: hover-provider-plugin/src/hover-provider-frontend-plugin.ts (hunk 2 above)
Comment (akurinnoy) [reply hover-provider-plugin/src/hover-provider-frontend-plugin.ts]: Thanks, I'll fix it
I'm preparing a PR for `che-generator-plugin` too.

File: hover-provider-plugin/yarn.lock (hunk 3 above)
Comment (akurinnoy) [reply hover-provider-plugin/yarn.lock]: Thanks, I'll substitute it with `@latest` version.

File: hover-provider-plugin/src/hover-provider-frontend-plugin.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*********************************************************************
+* Copyright (c) 2018 Oleksii Kurinnyi
```
Comment (benoitf) [inline hover-provider-plugin/src/hover-provider-frontend-plugin.ts]: it's not Red Hat ?

Comment (akurinnoy) [reply hover-provider-plugin/src/hover-provider-frontend-plugin.ts]: fixed

**Pull request merged**